src/h5web/dataset-visualizer/DatasetVisualizer.tsx

```diff
@@ -31,7 +33,13 @@ function DatasetVisualizer(props: Props): JSX.Element {
       <div className={styles.displayArea}>
         {dataset ? (
           activeVis && (
-            <VisDisplay key={dataset.id} vis={activeVis} dataset={dataset} />
+            <VisProvider
+              key={dataset.id}
```

Comment:
I wonder if the `key` is needed for the provider... I am not versed in these things so I would be happy to have your input.

Reply:
Remove it, open a 2D dataset, select another 2D dataset, observe if something behaves inappropriately while things update. You're looking for a component that still uses the data, dimension, domain, whatever... of the first dataset you selected while the second dataset is loading or its texture is being computed. For instance, make sure that: the domain in the toolbar changes instantly; the texture is not recomputed twice; etc. Try also with 1D datasets.

Reply:
I think it's still needed because there's still a render when the dimensions passed to the visualization (known right away from `dataset`) doesn't match the value retrieved with `useValue` (which is asynchronous). A fix might be to return a distinct `dataset` object from `useValue` asynchronously (i.e. by storing it with the value in the hook's local state). We can do this later.

Reply:
Agreed. I kept the key on `VisProvider` only.